nats.go

```diff
@@ -2382,7 +2383,12 @@ func (nc *Conn) waitForMsgs(s *Subscription) {
 
 		// Deliver the message.
 		if m != nil && (max == 0 || delivered <= max) {
-			mcb(m)
+			if jsi != nil && len(m.Data) == 0 && m.Header.Get(statusHdr) == flowControl {
```

Comment:
Note: should ideally be done after Ack

Reply:
Change flowControl to controlMsg and check reply for flowControl vs Heartbeat.